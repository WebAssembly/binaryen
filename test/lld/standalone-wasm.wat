(module
 (memory $0 2)
 (table $0 1 1 funcref)
 (elem (i32.const 0) $foo)
 (global $global$0 (mut i32) (i32.const 66112))
 (global $global$1 i32 (i32.const 66112))
 (global $global$2 i32 (i32.const 576))
 (export "memory" (memory $0))
 (export "main" (func $main))
 (export "__heap_base" (global $global$1))
 (export "__data_end" (global $global$2))
 (func $__original_main (result i32)
  (nop)
 )
 (func $main (param $0 i32) (param $1 i32) (result i32)
  (call $__original_main)
 )
 (func $foo (result i32))
)

