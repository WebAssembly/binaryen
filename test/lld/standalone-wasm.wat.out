(module
 (type $none_=>_i32 (func (result i32)))
 (type $i32_=>_i32 (func (param i32) (result i32)))
 (type $none_=>_none (func))
 (type $i32_=>_none (func (param i32)))
 (type $i32_i32_=>_i32 (func (param i32 i32) (result i32)))
 (memory $0 2)
 (table $0 1 1 funcref)
 (elem (i32.const 0) $foo)
 (global $global$0 (mut i32) (i32.const 66112))
 (global $global$1 i32 (i32.const 66112))
 (global $global$2 i32 (i32.const 576))
 (export "memory" (memory $0))
 (export "main" (func $main))
 (export "__heap_base" (global $global$1))
 (export "__data_end" (global $global$2))
 (export "stackSave" (func $stackSave))
 (export "stackAlloc" (func $stackAlloc))
 (export "stackRestore" (func $stackRestore))
 (export "__growWasmMemory" (func $__growWasmMemory))
 (export "_start" (func $_start))
 (func $__original_main (; 0 ;) (result i32)
  (nop)
 )
 (func $main (; 1 ;) (param $0 i32) (param $1 i32) (result i32)
  (call $__original_main)
 )
 (func $foo (; 2 ;) (result i32)
  (nop)
 )
 (func $stackSave (; 3 ;) (result i32)
  (global.get $global$0)
 )
 (func $stackAlloc (; 4 ;) (param $0 i32) (result i32)
  (local $1 i32)
  (global.set $global$0
   (local.tee $1
    (i32.and
     (i32.sub
      (global.get $global$0)
      (local.get $0)
     )
     (i32.const -16)
    )
   )
  )
  (local.get $1)
 )
 (func $stackRestore (; 5 ;) (param $0 i32)
  (global.set $global$0
   (local.get $0)
  )
 )
 (func $__growWasmMemory (; 6 ;) (param $newSize i32) (result i32)
  (memory.grow
   (local.get $newSize)
  )
 )
 (func $_start (; 7 ;)
  (drop
   (call $main
    (i32.const 0)
    (i32.const 0)
   )
  )
 )
)
(;
--BEGIN METADATA --
{
  "staticBump": 8,
  "tableSize": 1,
  "declares": [
  ],
  "externs": [
  ],
  "implementedFunctions": [
    "_main",
    "_stackSave",
    "_stackAlloc",
    "_stackRestore",
    "___growWasmMemory",
    "__start"
  ],
  "exports": [
    "main",
    "stackSave",
    "stackAlloc",
    "stackRestore",
    "__growWasmMemory",
    "_start"
  ],
  "namedGlobals": {
    "__heap_base" : "66112",
    "__data_end" : "576"
  },
  "invokeFuncs": [
  ],
  "features": [
  ],
  "mainReadsParams": 0
}
-- END METADATA --
;)
